src/ActionList/LinkItem.tsx

```diff
@@ -39,11 +39,24 @@ export const LinkItem = React.forwardRef(({sx = {}, active, as: Component, ...pr
     <Item
       active={active}
       sx={{paddingY: 0, paddingX: 0}}
-      _PrivateItemWrapper={({children, ...rest}) => (
-        <Link as={Component} sx={merge(styles, sx as SxProp)} {...props} {...rest} ref={forwardedRef}>
-          {children}
-        </Link>
-      )}
+      _PrivateItemWrapper={({children, onClick, ...rest}) => {
+        const clickHandler = (event: React.MouseEvent) => {
+          onClick && onClick(event)
+          props.onClick && props.onClick(event as React.MouseEvent<HTMLAnchorElement>)
+        }
+        return (
+          <Link
+            as={Component}
+            sx={merge(styles, sx as SxProp)}
+            {...rest}
+            {...props}
```

Comment:
The props passed in by the consumer should override the ones set by default by the item. Both onClicks will be called. We could do the same with onKeyPress, but consumers shouldn't really be using onKeyPress since it's deprecated for onKeyDown.